Helper/PaymentData.php

```diff
@@ -0,0 +1,312 @@
+            );
+        }
+
+        if ($paymentData instanceof PaymentOrderInterface) {
+            return $paymentData;
+        }
+
+        try {
+            $paymentData = $this->paymentQuoteRepo->getByPaymentIdPath($paymentIdPath);
+        } catch (NoSuchEntityException $e) {
+            $this->logger->error(
+                sprintf("Unable to find a SwedbankPay Payments Order with payment_id_path: %s", $paymentIdPath)
+            );
+        }
+
+        if ($paymentData instanceof PaymentQuoteInterface) {
+            return $paymentData;
+        }
+
+        $errorMessage = sprintf(
+            "Unable to find a SwedbankPay payment matching Payment ID Path:\n%s",
+            $paymentIdPath
+        );
+
+        $this->logger->error(
+            $errorMessage
+        );
+
+        throw new NoSuchEntityException(
+            new Phrase($errorMessage)
+        );
+    }
+
+    /**
+     * @param PaymentOrderInterface|PaymentQuoteInterface $payment
+     * @return bool
+     */
+    public function update($payment)
+    {
+        if ($payment instanceof PaymentOrderInterface) {
+            $this->paymentOrderRepo->save($payment);
+            return true;
+        }
+
+        if ($payment instanceof PaymentQuoteInterface) {
+            $this->paymentQuoteRepo->save($payment);
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * @param string $command
+     * @param string|int|float $amount
+     * @param PaymentOrderInterface|PaymentQuoteInterface $order
+     */
+    public function updateRemainingAmounts($command, $amount, $order)
+    {
+        switch ($command) {
+            case 'capture':
+                $order->setRemainingCapturingAmount($order->getRemainingCapturingAmount() - ($amount * 100));
+                $order->setRemainingCancellationAmount($order->getRemainingCapturingAmount());
+                $order->setRemainingReversalAmount($order->getRemainingReversalAmount() + ($amount * 100));
+                break;
+            case 'cancel':
+                $order->setRemainingCancellationAmount($order->getRemainingCancellationAmount() - ($amount * 100));
+                $order->setRemainingCapturingAmount($order->getRemainingCancellationAmount());
+                break;
+            case 'refund':
+                $order->setRemainingReversalAmount($order->getRemainingReversalAmount() - ($amount * 100));
+                break;
+        }
+
+        $this->paymentOrderRepo->save($order);
+    }
+
+    /**
+     * @param array $response
+     * @param string $instrument
+     */
+    public function saveQuoteToDB($response, $instrument)
+    {
+        /** @var MageQuote $mageQuote */
+        $mageQuote = $this->checkoutSession->getQuote();
+
+        $paymentIdPath = $response['payment']['id'];
+        $paymentId = $this->getSwedbankPayPaymentId($paymentIdPath);
```

Comment:
Why can't you just use `$paymentIdPath` directly? Why do you have to parse out the UUID from the URI instead of just storing the relative URI directly?